realtime_trains/extract.py

```diff
@@ -50,14 +74,49 @@ def get_realtime_trains_data(station_code: str) -> dict | None:
 
     station_data = get_data_from_api(api_url, username, password)
     if station_data:
-        logging.info("Data successfully retrieved.")
+        logging.info(
+            "Data successfully retrieved for station %s.", station_code)
         return station_data
-    logging.error("Failed to retrieve data from the API.")
+    logging.error(
+        "Extract: Failed to retrieve station %s from the API.", station_code)
     return None
 
 
+def get_all_stations_crs() -> list[str]:
+    '''Grabs all stations crs from the database'''
+
+    conn = get_connection()
+    cur = get_cursor(conn)
+
+    query = '''
+SELECT station_crs
+FROM station
+'''
+
+    cur.execute(query)
+    crs_data = cur.fetchall()
+    return [crs[0] for crs in crs_data]
```

Comment:
```suggestion
def get_all_stations_crs() -> list[str]:
    """Grabs all stations crs from the database"""
    conn = get_connection()
    with get_cursor(conn) as cur:
        cur.execute("SELECT station_crs FROM station")
        crs_data = cur.fetchone()
    return [crs[0] for crs in crs_data]
```

Reply:
Consider fetchone()

Reply:
Yeah that seems better, you also reminded me to add a close oops.